src/components/FullscreenModal/Footer/useStickyFooter.ts

```diff
@@ -0,0 +1,75 @@
+import { useCallback, useEffect, useState } from 'react';
+import debounce from 'lodash.debounce';
+
+export const useStickyFooter = (
+  modalRef: React.MutableRefObject<HTMLElement>,
+  modalFooterRef: React.MutableRefObject<HTMLElement>,
+): { isFixed: boolean; shouldShowScrollToTopButton: boolean } => {
+  const [isFixed, setIsFixed] = useState(false);
+  const [
+    shouldShowScrollToTopButton,
+    setShouldShowScrollToTopButton,
+  ] = useState(false);
+  const showScrollToTopButton = useCallback(() => {
+    if (modalRef.current === null) return;
+    const isScrollable =
+      modalRef.current.scrollHeight > modalRef.current.offsetHeight;
+    setShouldShowScrollToTopButton(isScrollable);
+    setIsFixed(false);
+  }, [modalRef, setShouldShowScrollToTopButton]);
```

Comment:
you can omit `setShouldShowScrollToTopButton` dependency, according to official docs: https://reactjs.org/docs/hooks-reference.html#usestate